src/main/java/org/osc/controller/nsc/api/SampleSdnRedirectionApi.java

```diff
@@ -495,4 +500,135 @@ public void close() throws Exception {
         });
     }
 
+    // Support methods for REST API's
+
+    public List<String> getInspectionHooksIds() throws Exception {
+
+        return this.txControl.supports(() -> {
+
+            List<InspectionHookEntity> elements = this.utils.txInspectionHookEntities();
+            List<String> elementList = new ArrayList<String>();
```

Comment:
you dont need the element list or the check for isEmpty

you could just do this.utils.txInspectionHookEntities().stream().map(InspectionHookEntity::getHookId).collect(Collectors.toList())